sources/api/pluto/src/api.rs

```diff
@@ -1,76 +1,101 @@
-    }
+#[derive(Deserialize)]
+struct SettingsView {
+    pub settings: View,
+}
 
-    /// Gets the info that we need to know about the EKS cluster from the Bottlerocket API.
-    pub(crate) async fn get_aws_k8s_info() -> Result<AwsK8sInfo> {
-        let settings = get_settings().await?;
-        Ok(AwsK8sInfo {
-            region: settings.aws.and_then(|a| a.region).map(|s| s.into()),
-            cluster_name: settings
-                .kubernetes
-                .as_ref()
-                .and_then(|k| k.cluster_name.clone())
-                .map(|s| s.into()),
-            cluster_dns_ip: settings.kubernetes.and_then(|k| k.cluster_dns_ip),
-        })
-    }
+#[derive(Debug, Snafu)]
+pub(crate) enum Error {
+    #[snafu(display("Failed to call apiclient: {}", source))]
+    CommandFailure { source: std::io::Error },
+    #[snafu(display("APIClient execution failed: {}", reason))]
+    ExecutionFailure { reason: String },
+    #[snafu(display("Deserialization of configuration file failed: {}", source))]
+    Deserialize {
+        #[snafu(source(from(serde_json::Error, Box::new)))]
+        source: Box<serde_json::Error>,
+    },
 }
 
-/// This dummy code is compiled when the `sources` workspace is being compiled for non `aws-k8s-*`
-/// variants.
-#[cfg(not(variant_family = "aws-k8s"))]
-mod inner {
-    use super::*;
-    use snafu::Snafu;
+pub(crate) async fn client_command<I, S>(args: I) -> Result<Vec<u8>>
+where
+    I: IntoIterator<Item = S>,
+    S: AsRef<OsStr>,
+{
+    let result = Command::new("/usr/bin/apiclient")
+        .args(args)
+        .output()
+        .await
+        .context(CommandFailureSnafu)?;
 
-    #[derive(Debug, Snafu)]
-    pub(crate) enum Error {
-        #[snafu(display(
-            "The get_aws_k8s_info function is only compatible with aws-k8s variants"
-        ))]
-        WrongVariant,
-    }
+    ensure!(
+        result.status.success(),
+        ExecutionFailureSnafu {
+            reason: String::from_utf8_lossy(&result.stderr)
+        }
+    );
+
+    Ok(result.stdout)
+}
 
-    pub(crate) async fn get_aws_k8s_info() -> Result<AwsK8sInfo> {
-        WrongVariantSnafu.fail()
-    }
+/// Gets the info that we need to know about the EKS cluster from the Bottlerocket API.
+pub(crate) async fn get_aws_k8s_info() -> Result<AwsK8sInfo> {
+    let view_str = client_command(&[
+        "get",
+        "settings.aws.region",
+        "settings.kubernetes.cluster-name",
+        "settings.kubernetes.cluster-dns-ip",
+    ])
+    .await?;
+    let view: SettingsView =
+        serde_json::from_slice(view_str.as_slice()).context(DeserializeSnafu)?;
+    Ok(AwsK8sInfo {
+        region: view.settings.aws.and_then(|a| a.region),
+        cluster_name: view
+            .settings
+            .kubernetes
+            .as_ref()
+            .and_then(|k| k.cluster_name.clone()),
+        cluster_dns_ip: view.settings.kubernetes.and_then(|k| k.cluster_dns_ip),
+    })
```

Comment:
One of the nice things about this refactor is that we can generate all the settings in one invocation, instead of calling `pluto` four times. That allows a more efficient implementation.

We can call `get_aws_k8s_info()` once and then pass `AwsK8sInfo` into each of the functions that needs it:
1. `get_cluster_dns_ip`
2. `get_node_ip` (also needs the cluster DNS IP, and can receive that as a parameter vs calling `get_cluster_dns_ip` again)
3. `get_private_dns_name`